core-backend/wasmtime/src/funcs.rs

```diff
@@ -92,12 +95,10 @@ impl<E: Ext + 'static> FuncsHandler<E> {
     pub fn exit_code(store: &mut Store<StoreData<E>>) -> Func {
         let f = move |caller: Caller<'_, StoreData<E>>| {
             let ext = &caller.data().ext;
-            let reply_tuple = ext.with(|ext: &mut E| ext.reply_to()).map_err(Trap::new)?;
-            if let Some((_, exit_code)) = reply_tuple {
-                Ok(exit_code)
-            } else {
-                Err(Trap::new("Not running in the reply context"))
-            }
+            ext.with(|ext: &mut E| ext.reply_to())
+                .and_then(|v| v.ok_or("Not running in the reply context"))
```

Comment:
should this be `with_fallible`?

Reply:
you see, there are only 2 functions (both in wasmtime), that return `Option`. So I see no big improvement in the new method. `with` is enough.